crates/ide_completion/src/render/function.rs

```diff
@@ -562,6 +562,78 @@ fn qux(Foo { bar }: Foo) {
 fn main() {
   qux(${1:foo})$0
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn complete_fn_param() {
+        // has mut kw
+        check_edit(
+            "mut bar",
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: (), mut ba$0)
+"#,
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: (), mut bar: u32)
+"#,
+        );
+
+        // has type param
+        check_edit(
+            "mut bar",
+            r#"
+fn g(foo: (), mut ba$0: u32)
+fn f(foo: (), mut bar: u32) {}
```

Comment:
Only when editor's insert mode set to `replace`? Is there a way to test edit results while lsp cap settings? Not seems possible with `CompletionConfig` yet